leaselock_test.go

```diff
@@ -0,0 +1,321 @@
+/*
+Copyright 2025 Swisscom (Schweiz) AG.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package leaselocker
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+
+	coordv1 "k8s.io/api/coordination/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func TestNewLeaseLock(t *testing.T) {
+	fakeCoordClient := &fakeLeaseClient{}
+
+	specs := []struct {
+		name             string
+		identity         string
+		nsn              types.NamespacedName
+		expectedDescribe string
+	}{
+		{
+			name:             "Valid NamespacedName",
+			identity:         "holder1",
+			nsn:              types.NamespacedName{Name: "test", Namespace: "ns"},
+			expectedDescribe: "ns/test",
+		},
+		{
+			name:             "Uppercase Name in NamespacedName",
+			identity:         "holder2",
+			nsn:              types.NamespacedName{Name: "TEST", Namespace: "ns"},
+			expectedDescribe: "ns/test",
+		},
+		{
+			name:             "Nil NamespacedName",
+			identity:         "holder3",
+			nsn:              types.NamespacedName{},
+			expectedDescribe: "/",
+		},
+	}
+
+	for _, tt := range specs {
+		t.Run(tt.name, func(t *testing.T) {
+			rlc := ResourceLockConfig{Identity: tt.identity}
+			got, err := newLeaseLock(tt.nsn, fakeCoordClient, rlc)
```

Comment:
```suggestion
			config := ResourceLockConfig{Identity: tt.identity}
			got, err := newLeaseLock(tt.nsn, fakeCoordClient, config)
```
Like this it's a bit easier/faster to see that the last variable is a configuraiton.